Library/Homebrew/brew.sh

```diff
@@ -390,8 +390,8 @@ fi
 check-run-command-as-root() {
   [[ "$(id -u)" = 0 ]] || return
 
-  # Allow Azure Pipelines/Docker/Kubernetes to do everything as root (as it's normal there)
-  [[ -f /proc/1/cgroup ]] && grep -E "azpl_job|docker|kubepods" -q /proc/1/cgroup && return
+  # Allow Azure Pipelines/Docker/Concourse/Kubernetes to do everything as root (as it's normal there)
+  [[ -f /proc/1/cgroup ]] && grep -E "azpl_job|docker|garden|kubepods" -q /proc/1/cgroup && return
```

Comment:
Maybe it's worth just removing the `grep` entirely assuming the `/proc/1/cgroup` doesn't exist on a non-container machine?

Reply:
(also: what is "Concourse"?)

Reply:
It's an "open-source continuous thing-doer": https://concourse-ci.org/

Reply:
Maybe. That'd be better than adding every CI provider people ever want to use. But was there a reason for adding the `grep` in the first place with only a few values?

Reply:
I don't know what's in `/proc/1/cgroup` on a typical, non-container Linux machine. Can you fill me in? The initial intent was just to stop this being a "all Linux machines" whitelist but we would like it to be an "all containers" whitelist.

Reply:
Looks like we do still need the `grep`, so this change will have to stay:

```
issyl0@vulpecula:~$ cat /proc/1/cgroup
12:cpuset:/
11:cpu,cpuacct:/
10:hugetlb:/
9:perf_event:/
8:pids:/
7:devices:/
6:net_cls,net_prio:/
5:rdma:/
4:memory:/
3:blkio:/
2:freezer:/
1:name=systemd:/init.scope
0::/init.scope
```

Reply:
I wonder if there's any better more generic version of detecting containers than this? @Homebrew/linux: thoughts?

Reply:
Wouldn't simple check for `/.dockerenv` file existence be sufficient? It seems to be present in every Docker container,

Reply:
I don't know myself of any easier way to detect if we're running in any container than simply extending this white list. There's also so many container environments, so I'm personally okay with continuing to extend this white list.

Reply:
Thanks. All my Googling turned up the same as we're already doing, too. Let's carry on like this.